src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -217,19 +217,36 @@ public static string GetLocationSafe(this Assembly a)
             }
         }
 
-        public static bool IsTypeIsInaccessibleForSerialization(Type t, Module currentModule, Assembly grainAssembly)
+        public static bool IsTypeIsInaccessibleForSerialization(Type t, Module fromModule, Assembly fromAssembly)
         {
-            if(t.GetCustomAttributes(typeof(SerializableAttribute), false).Length > 0)
```

Comment:
The reverse: returning `false` here means that the type does not have inaccessible types (which is a confusing double-negative) and will have a serializer generated for it. However, whether or not the type has the `[Serializable]` attribute doesn't affect whether or not it is accessible for serialization using the methods we use.